Opensds_state_machine_design.md

```diff
@@ -0,0 +1,109 @@
+# Opensds state machine design
+
+Status: WIP
+
+Version: Alpha
+
+Author(s): jack li (@lasshpanzer)
+
+## Summary
+
+The State machines of volume,snapshot and attachment show the real-time status and state transitions between actions.
+
+## Motivation
+
+For example, When creating a volume, there is needed to know if it has been created or if something went wrong.
```

Comment:
When creating a volume, there is a need to know if it has been created successfully or if something went wrong.

Remove "For example".